File: Python/Easy/Climbing-Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1 or n == 2:
            return n
        table = [0 for col in range(n)]
        table[0]=1
        table[1]=2
        for i in range(2,n):
            table[i] = table[i-1] + table[i-2]
        return table[n-1]
        
        

File: Java/Easy/Climbing-Stairs.java
class Solution {
    public int climbStairs(int n) {
        if (n == 1) {
            return 1;
        }
        if (n == 2) {
            return 2;
        }
        int[] table = new int[n];
        table[0] = 1;
        table[1] = 2;
        for (int i = 2; i<n; i++) {
            table[i] = table[i-1] + table[i-2];
        }
        return table[n-1];
    }
}
